src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -45,6 +47,17 @@ public ApplicationEngine(TriggerType trigger, IVerifiable container, StoreView s
             return disposable;
         }
 
+        internal bool TryAddUpdatedKey(StorageKey key)
+        {
+            bool keyAdded = false;
+            if (!updatedKeys.Contains(key))
+            {
+                updatedKeys.Add(key);
```

Comment:
It's better to return a value if it's added

Reply:
What do you mean? It returns true or false. Why should I return the added key if invoker already has it?

Reply:
not your method, updatedKeys.Add(key)

Reply:
But "Add" is void